diff --git a/gcc-wrapper.c b/gcc-wrapper.c
index cc110fa..0efb58d 100644
--- a/gcc-wrapper.c
+++ b/gcc-wrapper.c
@@ -253,6 +253,26 @@ static int finalize_i_files(char *file)
 
 	new_size_1 = trim_whitespaces(base_1, size_1);
 	new_size_2 = trim_whitespaces(base_2, size_2);
+	{
+		int fd;
+		long rv;
+
+		*sfx = 'a';
+		fd = open(file, O_CREAT | O_WRONLY | O_TRUNC, 0644);
+		if (fd >= 0) {
+			(void) (rv = write(fd, base_1, new_size_1));
+			close(fd);
+		}
+
+		*sfx = 'b';
+		fd = open(file, O_CREAT | O_WRONLY | O_TRUNC, 0644);
+		if (fd >= 0) {
+			(void) (rv = write(fd, base_2, new_size_2));
+			close(fd);
+		}
+
+		*sfx = '2';
+	}
 
 	buf = finalize_i_files_internal(base_1, (char *) base_1 + new_size_1,
 	                                base_2, (char *) base_2 + new_size_2);
@@ -387,10 +407,10 @@ static void produce_i_files(const char *const cc,
 
 	next:
 		if (*t == '2') {
-			unlink(ofile);
+			//unlink(ofile);
 			*t = '1';
 		}
-		unlink(ofile);
+		//unlink(ofile);
 
 		while (--p >= argv)
 			xfree(*p);
